src/main/scala/cromwell/engine/SymbolTable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding.{FullyQualifiedName, Call, WdlBinding, WdlValue}
+import scala.collection.mutable
+
+class SymbolTableEntry(val scope: String, val name: String, val iteration: Int, var wdlValue: Option[WdlValue], val wdlType: WdlType, input: Boolean) {
+  def isInput: Boolean = input
+
+  def isOutput: Boolean = !isInput
+}
+
+class SymbolTable(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]) {
+  val table = mutable.Set[SymbolTableEntry]()
+
+  binding.workflow.calls.foreach { call =>
+    call.task.inputs.foreach { case (inputName, wdlType) =>
+      table += new SymbolTableEntry(call.fullyQualifiedName, inputName, 0, None, wdlType, input = true)
+    }
+    call.task.outputs.foreach { output =>
+      table += new SymbolTableEntry(call.fullyQualifiedName, output.name, 0, None, output.wdlType, input = false)
+    }
+  }
+
+  inputs.foreach { case (fullyQualifiedName, value) =>
+    val lastIndex = fullyQualifiedName.lastIndexOf(".")
+    val (scope, name) = (fullyQualifiedName.substring(0, lastIndex), fullyQualifiedName.substring(lastIndex + 1))
+    val symbolTableEntry = table.find { entry => entry.scope == scope && entry.name == name }.get
+    symbolTableEntry.wdlValue = Option(value)
```

Comment:
So I gather that you're using the Option[WdlValue] to indicate that something may not be defined.  I have no objections to doing it that way.  Another approach might be to have a `WdlUndefined extends WdlValue` class.